lib/metrics/statsd.go

```diff
@@ -221,4 +240,23 @@ func (h *Statsd) Close() error {
 	return nil
 }
 
+// tags merges tag labels with their interpolated values
+// assumes that the input lists are the same length and
+// that labels[i] maps to values[i] for all i.
+//
+// Behavior for labels and values containing
+// the `:` character is undefined.
+func tags(labels []string, values []string) []string {
+	tags := make([]string, len(labels))
+	for i := range labels {
+		// We said we assumed the len(labels) == len(values),
+		// but we may as well check to be safe
+		if i >= len(values) {
```

Comment:
I think this check should be done at the start and if the length don't match this should not return an empty slice. Because there's no guarantee that the slices align up to the shorter of the 2 slices.

Reply:
Good call, updated.